### ISSUE TITLE
allow multiple files tabs, each tied to specific working directory

### ISSUE THREAD

**Issue by DrXyzzy** [sagemathinc/smc#1471]
This feature request addresses two things about current SMC that add clicks and differ from filesystem views (explorers) in other OSes and IDEs:

1. User is looking for something in a new path, i.e. one with no open files in SMC yet.

    - Click through file paths
    - Open a file
    - Decide it's not the file that is needed and close its tab
    - To view another file in the same directory, click search down the path has to be repeated
    - This leads to keeping files open just as a shortcut to get to their paths in the `Files` display
1. Working between two directories would be simpler if two tabs could be open, viewing files in each of the directories.

Allowing an SMC tab for a path, e.g. `~/COURSE-X/Week1`, would address these.

**Comment by DrXyzzy** [comment]
Mock-up screenshot:

![files-tabs](https://cloud.githubusercontent.com/assets/528072/21747641/5398f082-d533-11e6-9f3a-ff4f1b581d09.png)

**Comment by williamstein** [comment]
> if two tabs could be open,

HOW?

**Comment by DrXyzzy** [comment]
Oh, the `x` at right of the button for a file in the files bar could be used to dispose of a path link. So a two-attribute stateful pushpin icon isn't needed. It could be stateless.

**Comment by novoselt** [comment]
I am certainly going through the cycle Hal described, it would be very nice to resume in the same directory! In fact, when closing that file that was not the right one, I would like to be right away in the file explorer in the same folder, not in a different file tab. That is presumably implementable even without explorer being a real tab?..

**Comment by williamstein** [comment]
> That is presumably implementable even without explorer being a real tab?..

Yes.  Which behavior you want should likely be an account preference.
